internal/filter/expr/matcher.go

```diff
@@ -63,3 +63,31 @@ func Or[K any](matchers ...BoolExpr[K]) BoolExpr[K] {
 		return orMatcher[K]{matchers: matchers}
 	}
 }
+
+type andMatcher[K any] struct {
+	matchers []BoolExpr[K]
+}
+
+func (am andMatcher[K]) Eval(ctx context.Context, tCtx K) (bool, error) {
+	for i := range am.matchers {
```

Comment:
Any reason to use the index instead of the item returned, ie;

```golang
for _, match := range am 
```

Reply:
no, I was just following the existing convention in the file. I think the value loop is better than the way it's been done so I will change that for both.